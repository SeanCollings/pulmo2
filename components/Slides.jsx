import React, { useState } from 'react';
import { StyleSheet, ScrollView, View, Dimensions } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';

import { useTheme } from '../hooks/useTheme';

const SCREEN_WIDTH = Dimensions.get('window').width;

const SlidePosition = ({ totalSlides, selectedSlide, theme }) => (
  <View style={styles.slidePosition}>
    {[...Array(totalSlides).keys()].map((_, i) => (
      <MaterialCommunityIcons
        key={`sp-${i}`}
        name={i === selectedSlide ? 'circle' : 'circle-outline'}
        color={theme.BORDER}
        size={12}
      />
    ))}
  </View>
);

const Slides = ({ totalSlides, children }) => {
  const theme = useTheme();
  const [selectedSlide, setSelectedSlide] = useState(0);

  const onScrollHandler = (event) => {
    const offsetX = event.nativeEvent.contentOffset.x;
    setSelectedSlide(
      Math.floor(Math.floor(offsetX) / Math.floor(SCREEN_WIDTH))
    );
  };

  return (
    <View style={styles.scrollView}>
      <ScrollView
        horizontal
        pagingEnabled
        showsHorizontalScrollIndicator={false}
        onScroll={onScrollHandler}
        keyboardShouldPersistTaps="always"
      >
        {children}
      </ScrollView>
      <SlidePosition
        totalSlides={totalSlides}
        selectedSlide={selectedSlide}
        theme={theme}
      />
    </View>
  );
};

Slides.defaultProps = {
  data: [],
};

const styles = StyleSheet.create({
  scrollView: { flex: 1 },
  slidePosition: {
    flexDirection: 'row',
    justifyContent: 'center',
    position: 'absolute',
    bottom: 10,
    width: SCREEN_WIDTH,
  },
});

export default Slides;
