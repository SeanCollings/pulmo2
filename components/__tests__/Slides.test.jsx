import React from 'react';
import { act, create } from 'react-test-renderer';
import { cleanup } from 'react-native-testing-library';
import Slides from '../Slides';
import { Text } from 'react-native';

describe('Slides - unit test', () => {
  let component;

  const children = [
    <Text key={0}>one</Text>,
    <Text key={1}>two</Text>,
    <Text key={2}>three</Text>,
  ];

  afterEach(() => {
    cleanup();
  });

  test('should render', async () => {
    await act(async () => {
      component = create(
        <Slides children={children} totalSlides={children.length} />
      );
    });
    expect(component).toMatchSnapshot();
  });
});
