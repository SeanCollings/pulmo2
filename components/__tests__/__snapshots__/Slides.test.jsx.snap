// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slides - unit test should render 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    horizontal={true}
    keyboardShouldPersistTaps="always"
    onScroll={[Function]}
    pagingEnabled={true}
    showsHorizontalScrollIndicator={false}
  >
    <View>
      <Text>
        one
      </Text>
      <Text>
        two
      </Text>
      <Text>
        three
      </Text>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "bottom": 10,
        "flexDirection": "row",
        "justifyContent": "center",
        "position": "absolute",
        "width": 750,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#CCC",
            "fontSize": 12,
          },
          undefined,
          Object {
            "fontFamily": "material-community",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#CCC",
            "fontSize": 12,
          },
          undefined,
          Object {
            "fontFamily": "material-community",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#CCC",
            "fontSize": 12,
          },
          undefined,
          Object {
            "fontFamily": "material-community",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
